### PR TITLE
@craigspaeth Remove Mixpanel and Google Analytics in favor of Segment

### PR DESCRIPTION
**Pull request by wrgoldstein**
Addresses https://github.com/artsy/flare/issues/26.  Note I removed the analytics test both because the new analytics code is somewhat simpler and the test didn't seem to provide that much coverage.  If you'd be more comfortable with a test, let me know and I'll learn enough mocha to implement one.

### CODE REVIEW

**Comment by wrgoldstein** [comment]
This will require adding a `SEGMENT_API_KEY` to the environment where this is hosted, which I'm happy to do if you think this change is acceptable.

File: components/layout/bootstrap.coffee [hunk 1]
```diff
@@ -2,16 +2,15 @@ Backbone = require 'backbone'
...
 module.exports = ->
   require '../poplockit/jquery.poplockit.coffee'
 
 setupAnalytics = ->
-  # Initialize analytics & track page view if we included mixpanel
   # (not included in test environment).
-  return if not mixpanel? or mixpanel is 'undefined'
-  analytics(mixpanel: mixpanel, ga: ga)
-  analytics.trackPageview()
+  return if not analytics? or analytics is 'undefined'
```
Comment (craigspaeth) [inline components/layout/bootstrap.coffee]: The existential operator checks for null and undefined so you only need `return unless analytics?` here.

**Comment by craigspaeth** [comment]
LGTM 🙌  thank you! Minor style comment but not at all a blocker, just FYI.

File: components/layout/bootstrap.coffee (hunk 1 above)
Comment (wrgoldstein) [inline components/layout/bootstrap.coffee]: Got it, I was just blindly aping what was already there, thanks for letting me know.

**Comment by wrgoldstein** [comment]
Added the SEGMENT_API_KEY to flare-production in heroku.

**Pull request merged**